### PR TITLE
MM-15342 Use a smaller score post for certain screen sizes

### PR DESCRIPTION
**Pull request by hmhealey**
This uses the new, smaller design for the score post in the RHS and when there's not enough space in the centre channel because the RHS is open. Mobile view still falls back to the dropdown menu like before.

Screen Shot 2019-05-15 at 10 52 22 AM

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-15342

### CODE REVIEW

**Comment by hmhealey** [comment]
@deanwhillier I guess you must have the team sidebar visible when you made that screenshot? I forgot to include that in my calculations since my test account was only on one team before. I'll change the required dimensions slightly

**Comment by deanwhillier** [comment]
Yes, I'm using the sample data in my local install, so the team sidebar is showing.

**Comment by deanwhillier** [comment]
That works when the RHS is open!
Still seeing when the RHS is closed:
image

**Comment by hmhealey** [comment]
Oh, the other width that's used to remove the custom score post entirely is still off...

**Comment by hmhealey** [comment]
Also, compact mode actually has less horizontal space somehow. I'm not going to add a second set of widths for compact view, so I'll just increase the trigger widths to compensate

**Comment by hmhealey** [comment]
Re-requesting reviews since I've made some decently sized changes so that the size calculation now takes the team sidebar and compact view into consideration

**Comment by deanwhillier** [comment]
Getting some dropdown overlap for a very small width range with RHS open. 0/5 on doing anything about it.
Screen Shot 2019-05-23 at 1 08 09 PM

File: webapp/src/index.js [hunk 1]
```diff
@@ -12,46 +13,71 @@ export default class Plugin {
...
         registry.registerReducer(reducer);
 
+        // Register hooks
         const hooks = new Hooks(store);
         registry.registerMessageWillBePostedHook(hooks.messageWillBePosted);
 
+        // Register components
+        registry.registerRootComponent(Root);
+
+        window.addEventListener('resize', this.onWindowResize);
```
Comment (deanwhillier) [inline webapp/src/index.js]: We might want to throttle this event, thoughts? Might have missed that code if you already are.

File: webapp/src/selectors.js [hunk 2]
```diff
@@ -1,9 +1,82 @@
...
+export function getWindowWidth(state) {
+    return getPluginState(state).windowWidth;
+}
+
+export function isSurveyPostSmall(state, isRHS) {
+    return isRHS || !useFullSurveyPost(state);
+}
+
+function isCompactView(state) {
+    const compactView = get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, MESSAGE_DISPLAY, MESSAGE_DISPLAY_DEFAULT) === MESSAGE_DISPLAY_COMPACT;
```
Comment (deanwhillier) [inline webapp/src/selectors.js]: Could you maybe just return the result of this line and not worry about setting a variable?

File: webapp/src/selectors.js [hunk 3]
```diff
@@ -1,9 +1,82 @@
...
+function isSidebarOpen(state) {
+    return state.views.rhs.isSidebarOpen;
+}
+
+function isTeamSidebarVisible(state) {
+    return getMyTeams(state).length > 1;
+}
+
+// The required width for the regular and small survey posts with different combinations of display settings
+const requiredWidths = [
```
Comment (deanwhillier) [inline webapp/src/selectors.js]: For future consideration, we might want to brainstorm an alternative method for adjusting the UI to fit its container width as hardcoding specific pixel values can be brittle. 🙂

File: webapp/src/selectors.js (hunk 2 above)
Comment (hmhealey) [reply webapp/src/selectors.js]: Good catch. I think I must've been logging this value at some point

File: webapp/src/selectors.js (hunk 3 above)
Comment (hmhealey) [reply webapp/src/selectors.js]: Yeah, this isn't great. We could change the RHS to work like the team sidebar where the width is just subtracted from the total width required, but I didn't think of that until after working out these numbers. I'll leave that as something for future improvement

File: webapp/src/index.js (hunk 1 above)
Comment (hmhealey) [reply webapp/src/index.js]: I don't think we typically throttle the resize events since

**Comment by hmhealey** [comment]
@deanwhillier Would you mind filing a bug with the dropdown overlap? That dropdown is the general post action dropdown, and not part of the NPS plugin. I'd file one myself, but I don't know the screen width that you saw that at

**Comment by hmhealey** [comment]
@deanwhillier I'm going to merge this since there doesn't seem to be any outstanding issues that are caused by this PR

**Pull request merged**